### ISSUE TITLE
Erroneous no-inner-declarations

### ISSUE THREAD

**Issue by glenjamin** [babel/babel-eslint#1]
Great work on this! Not sure if these are eslint bugs or babel-eslint bugs, but reporting here for now.

No linter warning:
```js
/*eslint-env mocha*/

describe('stuff', function() {
    it("should help", function() {
        helper();
    });
    function helper() {
        return 1;
    }
});
```

Linter warning:
```js
/*eslint-env mocha*/

describe('stuff', () => {
    it("should help", () => {
        helper();
    });
    function helper() {
        return 1;
    }
});
```

**Comment by sebmck** [comment]
Fixed as of 1.0.8, thanks!

**Comment by btmills** [comment]
This was indeed a bug in ESLint, which I filed as eslint/eslint#1892 and fixed with eslint/eslint#1896. That fix will be included in the next release.

As we've gradually added support for ES6, we've encountered a number of issues around assumptions that were valid in ES5 but aren't any longer, and this was such a case. The goal is for ESLint to support arrow functions fully so that the workaround in 8a7b6b0 won't be necessary.
